src/pykeen/evaluation/evaluator.py

```diff
@@ -802,8 +827,12 @@ def _evaluate_batch(
     for unfiltered_evaluator in unfiltered_evaluators:
         if column == 2:  # tail scores
             process = unfiltered_evaluator.process_tail_scores_
-        else:
+        elif column == 1:  # relation scores
```

Comment:
Can we instead change the evaluator's interface to expose e.g. `_process_scores(keys=hrt_batch[:, :2], scores=scores, positive_mask=dense_positive_mask, side=SIDE_TAIL)` directly?

Reply:
There are some non-trivial things inside the different implementations of this in the rank-based evaluator and the classification evaluator, so for the moment I think the answer is no